### PR TITLE
Make `Merge` public again and rename it to `Merger`

### PR DESCRIPTION
**Pull request by llucax**
We need to make `Merge` public again, because the `Merge` receiver has more methods than a plain `Receiver`, in particular `stop()`, which users need to use for guaranteed cleanup.

But in this case it is better to introduce a breaking change to make sure users update to using the new `merge()` function, so we rename it to `Merger`. Also using a noun for the name is more in line with the other classes.

This also means now the `merge()` function returns the `Merger` instance instead of an abstract `Receiver`.

Additionally there are a couple of small unrelated cleanups:

* Remove `darglint`-related stuff, it was replaced by `pydoclint` a while ago.
* Remove cookiecutter template TODO comment.

### CODE REVIEW

**Comment by llucax** [comment]
Enabling auto-merge

File: src/frequenz/channels/_merge.py
Comment (daniel-zullo-frequenz) [inline src/frequenz/channels/_merge.py:18]: The `merge()` function returns the `Merge` instance instead of a `Receiver`. Should this be mentioned in the release notes?

**Review by daniel-zullo-frequenz — COMMENTED** [review]
I only have a question and LGTM otherwise

File: src/frequenz/channels/_merge.py
Comment (llucax) [reply src/frequenz/channels/_merge.py:18]: No, because `merge()` is unreleased so far.

**Comment by llucax** [comment]
@shsms last opportunity to complain, if @daniel-zullo-frequenz approves I will merge if you didn't complain before :laughing:

**Review by daniel-zullo-frequenz — APPROVED** [review]

**Pull request merged**
